src/ZenstruckFoundryBundle.php

```diff
@@ -294,6 +337,22 @@ public function process(ContainerBuilder $container): void
                 ->addMethodCall('addProvider', [new Reference($id)])
             ;
         }
+
+        // hooks
+        $iterators = [];
+        foreach ($container->findTaggedServiceIds('foundry.hook') as $id => $tags) {
+            $index = isset($tags[0]['class'])
+                ? HooksRegistry::hookClassSpecificIndex($tags[0]['hook'], $tags[0]['class'])
+                : $tags[0]['hook'];
+
+            $iterators[$index] ??= [];
+            $iterators[$index][] = new Reference($id);
+        }
```

Comment:
we're kinda missing a priority here. But that's not straightforward, because "global hook" and "hook specific to a class" are not in the same iterator.

I propose to leave this like this, and see if someone ever ask for it :)